frontend/src/lib/contexts/layout.ts

```diff
@@ -0,0 +1,68 @@
+import {setContext, getContext, hasContext} from 'svelte';
+import {deepMerge} from '$lib/utils/merge';
+import {stackedStore, type StackedStore} from '$lib/utils/stackedStore';
+import {error} from '@sveltejs/kit';
+
+const LAYOUT_CONTEXT_KEY = Symbol();
+
+export const DEFAULT_PAGE_STYLES: PageStyles = {
+  drawer: {
+    background: 'bg-base-100'
+  }
+} as const;
+
+/**
+ * Initialize and return the context. This must be called before `getLayoutContext()` and cannot be called twice.
+ * @returns The context object
+ */
+export function initLayoutContext(): LayoutContext {
+  if (hasContext(LAYOUT_CONTEXT_KEY)) error(500, 'InitLayoutContext() called for a second time');
+
+  const pageStyles = stackedStore<PageStyles, DeepPartial<PageStyles>>(
+    DEFAULT_PAGE_STYLES,
+    (current, value) => [
+      ...current,
+      deepMerge(structuredClone(current[current.length - 1]), structuredClone(value))
+    ]
+  );
+  // We can add more reversion actions here when needed
+  const revert = (index: number) => pageStyles.revert(index);
```

Comment:
But when we do, we just need to either use one stack with a more complex type for all changes or enforce consistent indexing between the stacks.